src/platform/site-wide/user-nav/containers/Main.jsx

```diff
@@ -96,6 +97,15 @@ export class Main extends React.Component {
     }
   };
 
+  handleSessionChange = event => {
+    if (!this.props.currentlyLoggedIn) return;
+
+    const { key, newValue } = event;
+    if (!key || (key === 'hasSession' && !newValue)) {
+      window.location.reload();
+    }
+  };
+
```

Comment:
Someone more qualified on the FE should scrutinize this for edge cases and potential logic flaws. 

@cvalarida @ncksllvn if either of you have time could you take a look please?